lib/schema-builder/factories/ast-definition-node.factory.ts

```diff
@@ -25,17 +26,32 @@ export class AstDefinitionNodeFactory {
   createObjectTypeNode(
     name: string,
     directiveMetadata?: DirectiveMetadata[],
-  ): ObjectTypeDefinitionNode | undefined {
-    if (isEmpty(directiveMetadata)) {
-      return;
-    }
+  ): ObjectTypeDefinitionNode {
```

Comment:
The check if a `extend` keyword should be printed by `printSchema` can be found here:

https://github.com/apollographql/apollo-server/blob/e44bb72baafbeaa90885ed7275a7df34c4183c98/packages/apollo-federation/src/service/printFederatedSchema.ts#L191-L192

As you can see, both the AST node and the extension AST node have to be set - so it is no longer possible to do an early bail on this one.

I am pretty sure removing this early bail does not break anything, but someone with more knowledge of the codebase should probably take a second look here.